librz/analysis/esil/esil_trace.c

```diff
@@ -124,21 +134,26 @@ static int trace_hook_reg_read(RzAnalysisEsil *esil, const char *name, ut64 *res
 		ret = esil->cb.reg_read(esil, name, res, size);
 	}
 	if (ret) {
-		ut64 val = *res;
-		//eprintf ("[ESIL] REG READ %s 0x%08"PFMT64x"\n", name, val);
-		sdb_array_add(DB, KEY("reg.read"), name, 0);
-		sdb_num_set(DB, KEYREG("reg.read", name), val, 0);
-	} //else {
-	//eprintf ("[ESIL] REG READ %s FAILED\n", name);
-	//}
+		// Trace reg read behavior
+		RzILTraceRegOp *reg_read = RZ_NEW0(RzILTraceRegOp);
+		reg_read->reg_name = strdup(name);
+		reg_read->behavior = TRACE_READ;
+		reg_read->value = *res;
+		esil_add_reg_trace(esil->trace, reg_read);
+	}
 	return ret;
 }
 
 static int trace_hook_reg_write(RzAnalysisEsil *esil, const char *name, ut64 *val) {
 	int ret = 0;
-	//eprintf ("[ESIL] REG WRITE %s 0x%08"PFMT64x"\n", name, *val);
-	sdb_array_add(DB, KEY("reg.write"), name, 0);
-	sdb_num_set(DB, KEYREG("reg.write", name), *val, 0);
+
+	// add reg write to trace
+	RzILTraceRegOp *reg_write = RZ_NEW0(RzILTraceRegOp);
+	reg_write->reg_name = strdup(name);
```

Comment:
IMO it would be better to use RzStrConstPool here

Reply:
It makes sense to find a better way handle const strings. Some const string from the argument are freed so I have to dump them manually. Where can I find the usage of ConstPool ?